File: android/src/main/java/com/reactnativelocalserver/utils/JSEvent.java
package com.reactnativelocalserver.utils;

import java.util.HashMap;
import java.util.Map;

public class JSEvent {
    private final String name;
    private final Map<String, String> body;

    public JSEvent(String name) {
        this.name = name;
        this.body = new HashMap();
        this.putString("type", name);
    }

    public String getName() {
        return name;
    }

    public Map<String, String> getBody() {
        return body;
    }

    public void putString(String key, String data) {
        body.put(key, data);
    }
    public void putInt(String key, int data) {
        body.put(key, String.valueOf(data));
    }
}

File: ios/LocalServer/Utils/JSEvent.swift
//
//  JSEvent.swift
//  LocalServer
//
//  Created by David Lang on 18.07.2022.
//  Copyright © 2022 Figure, Inc. All rights reserved.
//

class JSEvent {
    private let name: String
    private var body: Dictionary<String, Any> = [:]

    init(name: String) {
        self.name = name
        body["type"] = name
    }
    
    func putString(key: String!, value: String) -> Void {
        body[key] = value
    }
    
    func putInt(key: String!, value: UInt16) -> Void {
        body[key] = value
    }
    
    func getBody() -> Dictionary<String, Any> {
        return body
    }
    
    func getName() -> String {
        return name
    }
}
